client/quota.go

```diff
@@ -95,12 +97,9 @@ func (client *Client) RemoveQuotaGroup(groupName string) error {
 
 	var body bytes.Buffer
 	if err := json.NewEncoder(&body).Encode(data); err != nil {
-		return err
-	}
-	if _, err := client.doSync("POST", "/v2/quotas", nil, nil, &body, nil); err != nil {
-		return err
+		return "", err
 	}
-	return nil
+	return client.doAsync("POST", "/v2/quotas", nil, nil, &body)
```

Comment:
Nitpick, EnsureQuota above decorates the error from client.doAsync with extra context, should we have "cannot remove quota group: %w" here for consistency and better error report?

Reply:
seems a reasonable suggestion for a followup to me

Reply:
done in https://github.com/snapcore/snapd/pull/10484